File: src/main/java/com/company/erp/features/employees/dto/UpdateEmployeeRequest.java
package com.company.erp.features.employees.dto;

import com.company.erp.features.employees.entity.EmployeeStatus;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Size;

import java.time.LocalDate;

public record UpdateEmployeeRequest(
        Long userId,
        @Size(max = 64)  String employeeNo,
        @Size(max = 255) String fullName,
        @Email @Size(max = 255) String workEmail,
        @Size(max = 50)  String phone,
        @Size(max = 128) String position,
        @Size(max = 128) String department,
        LocalDate hireDate,
        LocalDate dateOfBirth,
        @Size(max = 16)   String gender,
        @Size(max = 1024) String address,
        @Size(max = 1024) String avatarUrl,
        @Size(max = 255)  String emergencyContact,
        @Size(max = 50)   String emergencyPhone,
        EmployeeStatus status
) {
}

File: app/Features/Employees/Requests/UpdateEmployeeRequest.php
<?php

namespace App\Features\Employees\Requests;

use App\Features\Employees\Models\EmployeeStatus;
use App\Support\Http\ApiFormRequest;
use Illuminate\Validation\Rule;

class UpdateEmployeeRequest extends ApiFormRequest
{
    public function rules(): array
    {
        return [
            'userId' => ['sometimes', 'nullable', 'integer'],
            'employeeNo' => ['sometimes', 'nullable', 'string', 'max:64'],
            'fullName' => ['sometimes', 'nullable', 'string', 'max:255'],
            'workEmail' => ['sometimes', 'nullable', 'email', 'max:255'],
            'phone' => ['sometimes', 'nullable', 'string', 'max:50'],
            'position' => ['sometimes', 'nullable', 'string', 'max:128'],
            'department' => ['sometimes', 'nullable', 'string', 'max:128'],
            'hireDate' => ['sometimes', 'nullable', 'date'],
            'dateOfBirth' => ['sometimes', 'nullable', 'date'],
            'gender' => ['sometimes', 'nullable', 'string', 'max:16'],
            'address' => ['sometimes', 'nullable', 'string', 'max:1024'],
            'avatarUrl' => ['sometimes', 'nullable', 'string', 'max:1024'],
            'emergencyContact' => ['sometimes', 'nullable', 'string', 'max:255'],
            'emergencyPhone' => ['sometimes', 'nullable', 'string', 'max:50'],
            'status' => ['sometimes', 'nullable', Rule::in(EmployeeStatus::values())],
        ];
    }
}
